.github/scripts/changed_files.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Get the list of changed files
+changed_files=$(git diff --name-only origin/$GITHUB_BASE_REF)
+
+# Define the directories to check
+allowed_directories=("docs/" "ui/" "website/" "grafana/" ".github/")
+
+# Initialize a variable to track directories outside the specified ones
+other_directories=""
+
+# Loop through the changed files and find directories outside the allowed ones
+for file in $changed_files; do
+  is_allowed=false
+  for dir in "${allowed_directories[@]}"; do
+    if [[ "$file" == "$dir"* ]] || [[ "$file" == *.md && "$dir" == *"/" ]]; then
+      is_allowed=true
+      break
+    fi
+  done
+  if [ "$is_allowed" = "false" ]; then
```

Comment:
I think this logic might be improved by changing to the below logic, since if one file outside of the skipped_directory is found we need to trigger the ci. This way we can exit the loop a little earlier.

```sh
trigger-ci = false
for file in $changed_files; do
    file_is_skipped = true
    for dir in "${skipped_directories[@]}"; do
            if [[ "$file" != "$dir"* ]] && [[ "$file" == *.md && "$dir" == *"/" ]]; then
                file_is_skipped =false
                break
            done
    done
    if file_is_skipped == false; do
        trigger-ci = true
        break
   done
done

if trigger-ci  == true; do
    trigger ci
else
    skip ci
```